qc_schema/dev/molecule.py

```diff
@@ -0,0 +1,99 @@
+    "properties": {
+        "symbols": {
+            "description": "The atom symbol for each atom in the molecule.",
+            "type": "array",
+            "items": {
+                "type": "string"
+            }
+        },
+        "geometry": {
+            "description": "The 3N XYZ coordinates of the atoms involved.",
+            "type": "array",
+            "items": {
+                "type": "number"
+            }
+        },
+        "masses": {
+            "description": "The masses of the atoms in the molecule, canonical weights assumed if not given.",
+            "type": "array",
+            "items": {
+                "type": "number"
+            }
+        },
+        "name": {
+            "description": "The name of the molecule.",
+            "type": "string"
+        },
+        "comment": {
+            "description": "Any additional comment one would attach to the molecule.",
+            "type": "string"
+        },
+        "charge": {
+            "description": "The overall charge of the molecule.",
+            "type": "number",
+            "default": 0.0
+        },
+        "multiplicity": {
+            "description": "The overall multiplicity of the molecule.",
+            "type": "number",
+            "multipleOf": 1.0,
+            "default": 1
+        },
+        "real": {
+            "description": "A list describing if the atoms are real or ghost.",
+            "type": "array",
+            "items": {
+                "type": "boolean"
+            }
+        },
+        "fragments": {
+            "description":
+            "A list of indices (0-indexed) for molecular fragments within the topology.",
+            "type":
+            "array",
+            "items": {
+                "type": "array",
+                "items": {
+                    "type": "number",
+                    "multipleOf": 1.0
+                }
+            }
+        },
+        "fragment_charges": {
+            "description": "A list of charges associated with the fragments tuple.",
+            "type": "array",
+            "items": {
+                "type": "number"
+            }
+        },
+        "fragment_multiplicities": {
+            "description": "A list of multiplicites associated with each fragment.",
+            "type": "array",
+            "items": {
+                "type": "number",
+                "multipleOf": 1.0
+            }
+        },
+        "fix_com": {
+            "description": "Whether to adjust to the molecule to the COM or not.",
+            "type": "boolean",
+            "default": False
+        },
+        "fix_orientation": {
+            "description": "Whether to rotate the molecule to a standard orientation or not.",
+            "type": "boolean",
+            "default": False
+        },
+        "provenance": {
+            "type": "object",
+            "$ref": "#/definitions/provenance"
+        }
+    },
+    "required": ["symbols", "geometry"],
```

Comment:
Is this actually enforced somehow?

Reply:
If you run this through a json-schema validator you will get a raise like the following:
```
E           jsonschema.exceptions.ValidationError: 'symbols' is a required property
E           Failed validating 'required' in schema['properties']['molecule']:
```

Im not sure how to require this at other levels.